packages/suite/src/actions/wallet/coinjoinAccountActions.ts

```diff
@@ -241,19 +242,83 @@ const coinjoinAccountCheckReorg =
             dispatch(transactionsActions.removeTransaction({ account, txs }));
         }
     };
 
 const coinjoinAccountAddTransactions =
-    (account: Account, transactions: ScanAccountProgress['transactions']) =>
-    (dispatch: Dispatch) => {
-        if (transactions.length) {
-            dispatch(transactionsActions.addTransaction({ account, transactions }));
+    (props: Parameters<typeof transactionsActions.addTransaction>[0]) => (dispatch: Dispatch) => {
+        if (props.transactions.length > 0) {
+            dispatch(transactionsActions.addTransaction(props));
         }
     };
 
+/**
+Action called from coinjoinMiddleware as reaction to prepending tx creation.
+Prepending tx could be created either as result of successful CoinjoinRound (not broadcasted by suite)
+or as result of sendFormActions > addFakePendingTxThunk (broadcasted by suite)
+in both cases Account should:
+- exclude spent utxo
+- mark addresses as used
+- recalculate anonymity
+- recalculate balance
+prepending txs have deadline (blockheigth) when they should be removed from UI
+ */
+export const updatePendingAccountInfo =
+    (accountKey: string) => async (dispatch: Dispatch, getState: GetState) => {
+        const state = getState();
+        const account = selectAccountByKey(state, accountKey);
+        const coinjoinAccount = selectCoinjoinAccountByKey(state, accountKey);
+        if (account?.backendType !== 'coinjoin' || !coinjoinAccount?.checkpoints) return;
+
+        const api = await dispatch(coinjoinClientActions.initCoinjoinService(account.symbol));
+        if (!api) return;
+
+        const { backend, client } = api;
+        const transactions = state.wallet.transactions.transactions[account.key];
+
+        const accountInfo = await backend.getAccountInfo(
+            account.descriptor,
+            transactions,
+            coinjoinAccount.checkpoints[0],
+            getAccountCache(account),
+        );
+
+        const { anonymityScores } = await client.analyzeTransactions(transactions, [
+            'anonymityScores',
+        ]);
+        accountInfo.addresses.anonymitySet = anonymityScores;
+
+        dispatch(accountsActions.updateAccount(account, accountInfo));
+    };
+
+export const createPendingTransaction =
+    (accountKey: string, payload: BroadcastedTransactionDetails) =>
+    async (dispatch: Dispatch, getState: GetState) => {
+        const state = getState();
+        const account = selectAccountByKey(state, accountKey);
+        const coinjoinAccount = selectCoinjoinAccountByKey(state, accountKey);
+        if (account?.backendType !== 'coinjoin' || !coinjoinAccount?.checkpoints) return;
+
+        const api = await dispatch(coinjoinClientActions.initCoinjoinService(account.symbol));
+        if (!api) return;
+
+        const { backend } = api;
+
+        const pending = await backend.createPendingTransaction(account, payload);
+        const deadline = state.wallet.blockchain[account.symbol].blockHeight + 2;
```

Comment:
why `+ 2`, please?

Reply:
if pending tx was not found in mempool after two mined blocks it should be removed, added some comment in code